pkg/types/encrypted_private_key.go

```diff
@@ -78,11 +79,11 @@ func (a *EncryptedPrivateKey) UnmarshalBinary(data []byte) error {
 }
 
 // Sign signs the given data using the private key.
-// password is destroyed just after being used even if an error occurs.
-func (e *EncryptedPrivateKey) Sign(password *memguard.LockedBuffer, salt, nonce, data []byte) ([]byte, error) {
+// Guarded password is destroyed.
+func (e *EncryptedPrivateKey) Sign(guardedPassword *memguard.LockedBuffer, salt, nonce, data []byte) ([]byte, error) {
```

Comment:
Why did you rename *password* to *guardedPassword*? The code readability has been made heavier without the information provided being important for the reader, because anyway you added an implementation comment explaining that the password will be destroyed.